### PR TITLE
sepolicy: Add sepolicy for USB gadget HAL

### PR DESCRIPTION
**Pull request by saranyagopal1**
Tracked-On: OAM-88508
Signed-off-by: saranya <saranya.gopal@intel.com>

### CODE REVIEW

File: usb-gadget/hal/file_contexts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# usb role switch service
+/sys/class/typec  						u:object_r:sysfs_usb_typec:s0
```
Comment (ceiba1985) [inline usb-gadget/hal/file_contexts]: Please move this line to genfs_contexts:
genfscon sysfs /class/typec               u:object_r:sysfs_usb_typec:s0

File: usb-gadget/hal/usb.te [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#rules for usb HAL
+type usb, domain;
+type usb_exec, exec_type, file_type, vendor_file_type;
+typeattribute usb hal_usb;
```
Comment (ceiba1985) [inline usb-gadget/hal/usb.te]: remove this line

File: usb-gadget/hal/usb.te [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+allow usb hal_usb_default_exec:file entrypoint;
+allow usb hwservicemanager:binder { transfer call };
+allow usb system_file:dir r_dir_perms;
+allow usb configfs:dir rw_dir_perms;
+allow usb configfs:file create_file_perms;
+allow usb vendor_usbconfig_prop:file { read open getattr map };
+allow usb self:netlink_kobject_uevent_socket { read bind create getopt setopt };
+allow usb sysfs_usb_typec:dir r_dir_perms;
+allow usb sysfs_usb_typec:file rw_file_perms;
+binder_call(system_server, usb)
```
Comment (ceiba1985) [inline usb-gadget/hal/usb.te]: a vendor domain cannot binder call a coredomain with treble enabled.

File: usb-gadget/hal/usbd.te [hunk 4]
```diff
@@ -0,0 +1 @@
+binder_call(usbd, usb)
```
Comment (ceiba1985) [inline usb-gadget/hal/usbd.te]: a core domain cannot binder call a vendor domain with treble enabled.

File: usb-gadget/hal/usb.te [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+hal_server_domain(usb, hal_usb)
+hal_server_domain(usb, hal_usb_gadget)
+init_daemon_domain(usb)
+
+allow usb hal_usb_default_exec:file entrypoint;
+allow usb hwservicemanager:binder { transfer call };
+allow usb system_file:dir r_dir_perms;
+allow usb configfs:dir rw_dir_perms;
+allow usb configfs:file create_file_perms;
+allow usb vendor_usbconfig_prop:file { read open getattr map };
```
Comment (ceiba1985) [inline usb-gadget/hal/usb.te]: replace this line with this:
get_prop(usb, vendor_usbconfig_prop)

File: usb-gadget/hal/file_contexts [hunk 6]
```diff
@@ -0,0 +1,3 @@
+# usb role switch service
+genfscon sysfs /sys/class/typec  				u:object_r:sysfs_usb_typec:s0
```
Comment (ceiba1985) [inline usb-gadget/hal/file_contexts]: it should be:
genfscon sysfs /class/typec  				u:object_r:sysfs_usb_typec:s0

and you need to move this line to genfs_contexts file.

**Review by ceiba1985 — APPROVED** [review]

**Review by kwang13 — APPROVED** [review]

**Pull request merged**